src/simulations/step.jl

```diff
@@ -37,6 +37,8 @@ function step_ahead!(model::ABM, agent_step!, model_step!, n, t)
     while until(t[], t0, n, model)
         !agents_first && model_step!(model)
         for id in schedule(model)
+            # ensure we don't act on agent that doesn't exist
+            hasid(model, id) || continue
```

Comment:
Note that this condition can be spared in the case of an ABM with a `Vector` container but we can also improve this in a separate PR